Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+                        //  --> so re-compute local x					
+                        local_y = carve::geom::cross(local_z, local_x);
+                        local_x = carve::geom::cross(local_y, local_z);
+
+                        // normalize the direction vectors
+                        local_x.normalize();
+                        local_y.normalize();
+                        local_z.normalize();
+
+                        // produce a quaternion
+                        matrix = carve::math::Matrix(
+                            local_x.x, local_y.x, local_z.x, translate.x,
+                            local_x.y, local_y.y, local_z.y, translate.y,
+                            local_x.z, local_y.z, local_z.z, translate.z,
+                            0, 0, 0, 1);
+                    }
+
+                    // The implementation is in file PlacementConverterImpl.h
+                    std::shared_ptr<typename IfcEntityTypesT::IfcCurve> GetCurveOfPlacement(
+                        const std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement>& linearPlacement);
+
+
+
+                    /*! \brief Retrieves the respective placement type from the \c IfcLocalPlacement and converts it.
+
+                    \param[in]	    local_placement		\c IfcLocalPlacement entity to be interpreted.
+                    \param[in,out]  matrix			Calculated transformation matrix.
+
+                    Retrieves the respective placement type from the \c IfcLocalPlacement and converts it.
+                    The result is stored in matrix. Uses the switch statement to get the correct type.
+                    */
+                    void convertRelativePlacement(carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLocalPlacement> local_placement)
+                    {
+                        decltype(local_placement->RelativePlacement)& axis2placement = local_placement->RelativePlacement;
+                        switch(axis2placement.which()) {
+                        case 0:
+                            convertIfcAxis2Placement2D(axis2placement.get<0>().lock(), object_placement_matrix);
+                            break;
+                        case 1:
+                            convertIfcAxis2Placement3D(axis2placement.get<1>().lock(), object_placement_matrix);
+                            break;
+                        default:
+                            BLUE_LOG(fatal) << local_placement->getErrorLog() << "RelativePlacement conversion issues.";
+                            break;
+                        }
+                    }
+
+                    void convertIfcLocalPlacementRelationPoint(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, std::shared_ptr<typename IfcEntityTypesT::IfcLocalPlacement> local_placement, carve::math::Matrix &relative_placement)

```

Comment:
Also, change the out parameter to return value.

Reply:
Also done, as depends on IfcObjectPlacement.